encoding/thrift/inbound_nowire.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import (
+	"context"
+	"io"
+
+	"go.uber.org/thriftrw/protocol/binary"
+	"go.uber.org/thriftrw/protocol/stream"
+	"go.uber.org/thriftrw/wire"
+	encodingapi "go.uber.org/yarpc/api/encoding"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/pkg/errors"
+)
+
+var _emptyResponse = NoWireResponse{}
+
+type thriftNoWireHandler struct {
+	NoWireHandler NoWireHandler
+	Protocol      stream.Protocol
+	Enveloping    bool
+}
+
+type NoWireCall struct {
+	Reader        io.Reader
+	RequestReader stream.RequestReader
+	EnvelopeType  wire.EnvelopeType
+
+	StreamReader stream.Reader
+}
+
+type NoWireHandler interface {
+	Handle(context.Context, *NoWireCall) (NoWireResponse, error)
+}
```

Comment:
What is the reason we did not follow the same pattern as:
```
// UnaryHandler is a convenience type alias for functions that act as Handlers.
type UnaryHandler func(context.Context, wire.Value) (Response, error)

// OnewayHandler is a convenience type alias for functions that act as OnewayHandlers.
type OnewayHandler func(context.Context, wire.Value) error
```

I would prefer to have the same pattern here for consistency between the nowire and wire implementation (easy to maintain). It will also simplify the way the handler are being created in the thriftrw yarpc plugin and how it is register (from https://github.com/yarpc/yarpc-go/pull/2085)